web/files/files.go

```diff
@@ -19,54 +22,159 @@ import (
 // DefaultContentType is used for files uploaded with no content-type
 const DefaultContentType = "application/octet-stream"
 
-// Upload is the handler for uploading a file
-//
-// swagger:route POST /files/:folder-id files uploadFile
+// DocType is the type of document, eg. file or folder
+type DocType string
+
+const (
+	// FileDocType is document type
+	FileDocType DocType = "io.cozy.files"
+	// FolderDocType is document type
+	FolderDocType = "io.cozy.folders"
+)
+
+var (
+	errDocAlreadyExists = errors.New("Directory already exists")
+	errDocTypeInvalid   = errors.New("Invalid document type")
+	errIllegalFilename  = errors.New("Invalid filename: empty or contains one of these illegal characters: / \\ : ? * \" |")
```

Comment:
If we don't want to enforce these rules, I don't think its necessary to forbid these special characters either. I'll just leave `/` as an illegal one for now.

New PR: #30